pkg/controllers/nodeclaim/podevents/controller.go

```diff
@@ -81,14 +78,28 @@ func (c *Controller) Reconcile(ctx context.Context, pod *corev1.Pod) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
-	// If we've set the lastPodEvent before and it hasn't been before the timeout, don't do anything
-	if !nc.Status.LastPodEventTime.Time.IsZero() && c.clock.Since(nc.Status.LastPodEventTime.Time) < dedupeTimeout {
+	var eventTime time.Time
+	// If pod is being removed (terminal or terminating), use current time
+	if podutils.IsTerminal(pod) || podutils.IsTerminating(pod) {
+		eventTime = c.clock.Now()
+	} else {
+		// Otherwise check for PodScheduled condition
+		if cond, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
+			return c.Type == corev1.PodScheduled && c.Status == corev1.ConditionTrue
+		}); ok {
+			eventTime = cond.LastTransitionTime.Time
+		}
+	}
+
+	// If we don't have a valid time, skip
+	if eventTime.IsZero() {
 		return reconcile.Result{}, nil
 	}
 
 	// otherwise, set the pod event time to now
 	stored := nc.DeepCopy()
-	nc.Status.LastPodEventTime.Time = c.clock.Now()
+
+	nc.Status.LastPodEventTime.Time = eventTime
```

Comment:
This needs to be max otherwise it's going to oscillate around as different pods get reconciled. Also, I think still applying a dedupe timeout makes sense -- basically, if the current stored time is within (say 10s) of the new time, then don't update it

Reply:
ammended